ci/prow/make_config.go

```diff
@@ -699,6 +773,46 @@ func strExists(arr []string, str string) bool {
 	return false
 }
 
+// executeJobTemplateWrapper takes in consideration of repo settings, decides how many varianats of the
+// same job needs to be generated and generates them.
+func executeJobTemplateWrapper(repoName string, data interface{}, generateOneJob func(data interface{})) {
+	var legacyBranches []string
+	// Find out if LegacyBranches is set in repo settings
+	for _, repo := range repositories {
+		if repo.Name == repoName {
+			if len(repo.LegacyBranches) > 0 {
+				legacyBranches = repo.LegacyBranches
+			}
+		}
+	}
+	if len(legacyBranches) == 0 { // Generate only one job as normal if LegacyBranches is not set
+		generateOneJob(data)
+	} else {
+		// Generate one job with 'knative.dev' path alias for branches other than legacy branches,
+		// and another job without path alias for legacy branches
+		var base *baseProwJobTemplateData
+		switch v := data.(type) {
+		case *presubmitJobTemplateData:
+			base = &data.(*presubmitJobTemplateData).Base
+		case *postsubmitJobTemplateData:
+			base = &data.(*postsubmitJobTemplateData).Base
+		default:
+			log.Fatalf("Unrecognized job template type: '%v'", v)
+		}
+		branches := base.Branches
+		skipBranches := base.SkipBranches
+		base.PathAlias = ""
+		base.Branches, base.SkipBranches = consolidateBranches(branches, skipBranches, legacyBranches, make([]string, 0))
+		generateOneJob(data)
+		base.Branches, base.SkipBranches = consolidateBranches(branches, skipBranches, make([]string, 0), legacyBranches)
+		base.PathAlias = "path_alias: knative.dev/" + base.RepoName
+		base.ExtraRefs = append(base.ExtraRefs, "  "+base.PathAlias)
+		generateOneJob(data)
+		base.Branches = branches
```

Comment:
`PathAlias` and `ExtraRefs` are modified and not restored, why bothering with `Branches` and `SkipBranches`? AFAICU this is not called in a loop for the same data.

Reply:
that's right, removed